central/platform/reprocessor/reprocessor_impl.go

```diff
@@ -76,46 +75,7 @@ func (pr *platformReprocessorImpl) runReprocessing() {
 	}
 }
 
-func (pr *platformReprocessorImpl) alertsNeedReprocessing() (bool, error) {
-	// Check if there is atleast one alert where platform component is unset
-	q := unsetPlatformComponentQuery.CloneVT()
-	q.Pagination = &v1.QueryPagination{
-		Limit: 1,
-	}
-	alerts, err := pr.alertDatastore.GetByQuery(reprocessorCtx, q)
```

Comment:
Originally my thought was to run a count query in these `needReprocessing` functions but there was a bug in Count function (now fixed by #13778 ) which led me to use `GetByQuery`. But you are right, the batch processing loop is already structured in a way that there is no need for this check.

Reply:
Initially I also thought about Count but I was not sure what the query looks like and once I found it's reused I thought let's remove it